### PR TITLE
Refactor MSA name strings to output with the desired format

### PR DESCRIPTION
**Pull request by m3brown**
Pull MSAMD data from [Qu API's msamd.csv](https://github.com/cfpb/api/blob/master/resources/datasets/hmda/code_sheets/msamd.csv) or something similarly formatted.

Create a list of MSAMDs and update the list in src/static/js/templates/location.html.
 - Include all MSAs, even if they are not present in the latest year
 - Use the most recent name for each MSA
 - If an MSA is not used for all years, put the applicable date range in the string

Usage:
```shell
# pull from github.com/cfpb/api, print updated HTML content to the terminal
./generate_msamd_dropdown.py

# instead of printing to the terminal, swap the results into location.html (after backing it up)
./generate_msamd_dropdown.py -o location.html

# override the default CSV source with a URL
./generate_msamd_dropdown.py -u http://github.com/m3brown/api/raw/master/resources/datasets/hmda/code_sheets/msamd.csv

# override the default CSV source with a file
./generate_msamd_dropdown.py -f ../mypath/msamd.csv

```

### CODE REVIEW

File: src/static/js/templates/generate_msamd_dropdown.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    else:
+        csv_url = args.url
+    csv_feed = urlopen(csv_url)
+elif args.file:
+    csv_feed = open(args.file)
+else:
+    csv_feed = urlopen(DEFAULT_URL)
+
+
+def rangeify(years_list):
```
Comment (sleitner) [inline src/static/js/templates/generate_msamd_dropdown.py]: I didn't dig in, but an unsorted list gives me a different answer from a sorted one. (This should probably get a unit test.)

File: src/static/js/templates/generate_msamd_dropdown.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+if '99999' in codes:
+    codes.remove('99999')
+all_years = set([int(line['year']) for line in csv_content])
+
+
+# Print HTML with latest name. Also, if the code is not present for all years,
+# print the list of years after the name.
+html_msa_list = []
+for code in codes:
+    code_filter = [row for row in csv_content if row['code'] == code]
```
Comment (sleitner) [inline src/static/js/templates/generate_msamd_dropdown.py]: nice!

File: src/static/js/templates/generate_msamd_dropdown.py (hunk 1 above)
Comment (m3brown) [inline src/static/js/templates/generate_msamd_dropdown.py]: Good catch, fixed.

Also, I agree about unit tests, unfortunately this script isn't very unit-test friendly as it's written.

**Comment by sleitner** [comment]
Nice job!

**Pull request merged**